test/testPriceRouter/PriceRouter.t.sol

```diff
@@ -0,0 +1,788 @@
+    }
+
+    function testEditAsset() external {
+        PriceRouter.AssetSettings memory settings = PriceRouter.AssetSettings(EXTENSION_DERIVATIVE, address(this));
+
+        // Trying to add an existing asset should revert.
+        vm.expectRevert(
+            bytes(abi.encodeWithSelector(PriceRouter.PriceRouter__AssetAlreadyAdded.selector, address(USDC)))
+        );
+        priceRouter.addAsset(USDC, settings, abi.encode(0), 1e8);
+
+        // Trying to edit some new asset should revert.
+        vm.expectRevert(bytes(abi.encodeWithSelector(PriceRouter.PriceRouter__AssetNotAdded.selector, address(777))));
+        priceRouter.startEditAsset(ERC20(address(777)), settings, abi.encode(0));
+
+        // USDC is already added, edit it to use this contract as an extension.
+        bytes32 editHash = keccak256(abi.encode(USDC, settings, abi.encode(0)));
+        priceRouter.startEditAsset(USDC, settings, abi.encode(0));
+
+        assertEq(
+            priceRouter.assetEditableTimestamp(editHash),
+            block.timestamp + priceRouter.EDIT_ASSET_DELAY(),
+            "Asset editable timestamp should be current time plus edit delay."
+        );
+
+        // Owner calling `completeEditAsset` with an asset not pending edit should revert.
+        vm.expectRevert(
+            bytes(abi.encodeWithSelector(PriceRouter.PriceRouter__AssetNotEditable.selector, address(777)))
+        );
+        priceRouter.completeEditAsset(ERC20(address(777)), settings, abi.encode(0), 1e8);
+
+        vm.warp(block.timestamp + priceRouter.EDIT_ASSET_DELAY() / 2);
+
+        // Owner calling `completeEditAsset` early should revert.
+        vm.expectRevert(
+            bytes(abi.encodeWithSelector(PriceRouter.PriceRouter__AssetNotEditable.selector, address(USDC)))
+        );
+        priceRouter.completeEditAsset(USDC, settings, abi.encode(0), 1e8);
+
+        // Once enough time has passed, then `completeEditAsset` will work.
+        vm.warp(block.timestamp + priceRouter.EDIT_ASSET_DELAY());
+        priceRouter.completeEditAsset(USDC, settings, abi.encode(0), 1e8);
+
+        assertEq(priceRouter.assetEditableTimestamp(editHash), 0, "Asset editable timestamp should be zero.");
+
+        // Trying to cancel an edit asset when no edit is pending reverts.
+        vm.expectRevert(
+            bytes(abi.encodeWithSelector(PriceRouter.PriceRouter__AssetNotPendingEdit.selector, address(USDC)))
+        );
+        priceRouter.cancelEditAsset(USDC, settings, abi.encode(0));
+
+        // If an asset is pending edit, it can be cancelled.
+        priceRouter.startEditAsset(USDC, settings, abi.encode(0));
+
+        priceRouter.cancelEditAsset(USDC, settings, abi.encode(0));
+    }
+
+    // ======================================= PRICING TESTS =======================================
+
+    function testExchangeRate() external {
+        PriceRouter.AssetSettings memory settings = PriceRouter.AssetSettings(CHAINLINK_DERIVATIVE, BOND_ETH_FEED);
```

Comment:
Is it worthwhile to test the various different derivative types, or is that captured in Extension tests more-so?

Reply:
Not really, this test is verifying that our math in getExchangeRate is correct, so it is independent from actually seeing if how our derivative is calculating price is correct.